openoa/utils/downloader.py

```diff
@@ -333,6 +336,248 @@ def get_era5_monthly(
+    save_filename: str,
+    start_date: str = "2000-01",
+    end_date: str = None,
+    calc_derived_vars: bool = False,
+) -> pd.DataFrame:
+    """
+    Get ERA5 data directly from the CDS service. This requires registration on the CDS service.
+    See registration details at: https://cds.climate.copernicus.eu/api-how-to
+
+    This function returns hourly ERA5 data from the "ERA5 monthly averaged data on single levels
+    from 1959 to present" dataset. See further details regarding the dataset at:
+    https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means.
+    U and V components of wind speed at 100 m, temperature at 2 m, and surface pressure are
+    downloaded here.
+
+    As well as returning the data as a dataframe, the data is also saved as monthly NetCDF files and
+    a csv file with the concatenated data. These are located in the "save_pathname" directory, with
+    "save_filename" prefix. This allows future loading without download from the CDS service.
+
+    Args:
+        lat(:obj:`float`): Latitude in WGS 84 spatial reference system (decimal degrees).
+        lon(:obj:`float`): Longitude in WGS 84 spatial reference system (decimal degrees).
+        save_pathname(:obj:`str` | :obj:`Path`): The path where the downloaded reanalysis data will
+            be saved.
+        save_filename(:obj:`str`): The file name used to save the downloaded reanalysis data.
+        start_date(:obj:`str`): The starting year, month, and day that data is downloaded for. This
+            should be provided as a string in the format "YYYY-MM-DD". Defaults to "2000-01-01".
+        end_date(:obj:`str`): The final year, month, and day that data is downloaded for. This should be
+            provided as a string in the format "YYYY-MM-DD". Defaults to current date. Note that data
+            may not be available yet for the most recent couple months.
+        calc_derived_vars (:obj:`bool`, optional): Boolean that specifies whether wind speed, wind
+            direction, and air density are computed from the downloaded reanalysis variables and
+            saved. Defaults to False.
+
+
+    Returns:
+        df(:obj:`dataframe`): A dataframe containing time series of the requested reanalysis
+            variables:
+            1. u_ms: the U component of wind speed at a height of 100 m in m/s.
+            2. v_ms: the V component of wind speed at a height of 100 m in m/s.
+            3. temperature_K: air temperature at a height of 2 m in Kelvin.
+            4. surf_pres_Pa: surface pressure in Pascals.
+
+    Raises:
+        ValueError: If the start_date is greater than the end_date.
+        Exception: If unable to connect to the cdsapi client.
+    """
+
+    logger.info("Please note access to ERA5 data requires registration")
+    logger.info("Please see: https://cds.climate.copernicus.eu/api-how-to")
+
+    # set up cds-api client
+    try:
+        c = cdsapi.Client()
+    except Exception as e:
```

Comment:
Is this what's raised from the API? I thought it was more specific for some reason, or are you just trying to capture any failed connection, regardless of reason?